webapp/src/component/layout/Notifications/NotificationsPopup.tsx

```diff
@@ -170,13 +141,25 @@ export const NotificationsPopup: FunctionComponent<{
         <ListItemHeader divider>
           <T keyName="notifications-header" />
         </ListItemHeader>
-        {notifications?.map((notification, i) => (
-          <NotificationItem
-            notification={notification}
-            key={notification.id}
-            isLast={i === notifications.length - 1}
-          />
-        ))}
+        {notifications?.map((notification, i) => {
+          const props = {
+            notification: notification,
+            key: notification.id,
+            isLast: i === notifications.length - 1,
+          };
+          switch (notification.type) {
```

Comment:
This should also go out. It can be a Record<Type, Component with specific props>. Then you can just find it in the map.